azureml_scripts/make_deployment.py

```diff
@@ -0,0 +1,147 @@
+import tempfile
+from argparse import ArgumentParser, Namespace
+from contextlib import contextmanager
+from pathlib import Path
+from logging import Logger
+
+import requests
+from azure.ai.ml import MLClient
+from azure.ai.ml.entities import (
+    Environment,
+    ManagedOnlineDeployment,
+    ManagedOnlineEndpoint,
+    Model,
+)
+from azure.identity import DefaultAzureCredential
+
+
+def main() -> None:
+    module = "azureml-model-deployment"
+    logger = Logger(module)
+    args = parse_args()
+
+    logger.info(f"{module}: Creating Azure MLClient...")
+
+    # Create a AzureML client
+    ml_client = MLClient(
+        DefaultAzureCredential(),
+        args.subscription_id,
+        args.resource_group,
+        args.workspace,
+    )
+
+    # Create an endpoint
+    logger.info(f"{module}: Getting or creating endpoint...")
+    endpoint = _get_or_create_endpoint(ml_client, args.endpoint_name)
+
+    # Create a deployment environment
+    logger.info(f"{module}: Creating deployment environment...")
+    env = _create_deployment_env(args.image, args.env_name)
+
+    # Create the deployment
+    logger.info(f"{module}: Deploying to endpoint...")
+    _create_or_update_deployment(
+        ml_client,
+        args.endpoint_name,
+        args.deployment_name,
+        env,
+    )
+
+    # Route all traffic to the deployment
+    logger.info(f"{module}: Routing all traffic to current deployment...")
+    endpoint = _route_all_traffic_to_deployment(
+        ml_client, endpoint, args.deployment_name
+    )
+
+    # Get the API key
+    logger.info(f"{module}: Getting api-key and testing endpoint...")
+    api_key = ml_client.online_endpoints.get_keys(args.endpoint_name).primary_key
+    # Make a request to the endpoint to test it
+    url = f"https://{args.endpoint_name}.westeurope.inference.ml.azure.com"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {api_key}",
+        # 'azureml-model-deployment': deployment_name # Can be used to route traffic to a specific deployment if it doesn't have 100% traffic
+    }
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+    print(response.json())
+    logger.info(f"{module} Success!")
+
+
+def _get_or_create_endpoint(ml_client, endpoint_name):
+    def _does_endpoint_exist(ml_client: MLClient, endpoint_name: str):
```

Comment:
All the args are passed to this function. Is there any reason why this is a nest function?
I would either change it into a statement or a normal function